### PR TITLE
Update links in Types of Cross-Site Scripting

### PR DESCRIPTION
**Pull request by adamczi**
Updated
* relative links for attack pages
* absolute links for cheat sheets and PDF resources
* create local copy and use the following image - https://www.linuxsecrets.com/owasp-wiki/images/c/c6/Server-XSS_vs_Client-XSS_Chart.jpg

### CODE REVIEW

File: pages/Types_of_Cross-Site_Scripting.md [hunk 1]
```diff
@@ -16,13 +16,13 @@ This article describes the many different types or categories of
 cross-site scripting (XSS) vulnerabilities and how they relate to each
 other.
 
-Early on, two primary types of [XSS](XSS "wikilink") were identified,
+Early on, two primary types of [XSS](../attacks/Cross-site-Scripting-(XSS)) were identified,
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:19]: This doesn't exist:
https://owasp.org/www-community/attacks/Cross-site-Scripting-(XSS)

**Comment by kingthorin** [comment]
Images should be stored in `assets/images`.

Please ensure the image has an appropriate license before including it.

File: pages/Types_of_Cross-Site_Scripting.md (hunk 1 above)
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:19]:
```suggestion
Early on, two primary types of [XSS](attacks/xss/) were identified,
```

For: https://owasp.org/www-community/attacks/xss/

File: pages/Types_of_Cross-Site_Scripting.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ This article describes the many different types or categories of
...
 Stored XSS and Reflected XSS. In 2005, [Amit Klein defined a third type
 of XSS](http://www.webappsec.org/projects/articles/071105.shtml), which
-he coined [DOM Based XSS](DOM_Based_XSS "wikilink"). These 3 types of
+he coined [DOM Based XSS](../attacks/DOM_Based_XSS). These 3 types of
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:22]:
```suggestion
he coined [DOM Based XSS](attacks/DOM_Based_XSS). These 3 types of
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ can envision the attack payload being permanently stored in the victim’s
 browser, such as an HTML5 database, and never being sent to the server
 at all.
 
-### [Reflected XSS](Cross-site_Scripting_\(XSS\)#Reflected_XSS_Attacks "wikilink") (AKA Non-Persistent or Type II)
+### [Reflected XSS](../attacks/Cross-site-Scripting-(XSS)#reflected-xss-attacks) (AKA Non-Persistent or Type II)
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:36]:
```suggestion
### [Reflected XSS](attacks/xss/#reflected-xss-attacks) (AKA Non-Persistent or Type II)
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 4]
```diff
@@ -16,13 +16,13 @@ This article describes the many different types or categories of
...
-Early on, two primary types of [XSS](XSS "wikilink") were identified,
+Early on, two primary types of [XSS](../attacks/Cross-site-Scripting-(XSS)) were identified,
 Stored XSS and Reflected XSS. In 2005, [Amit Klein defined a third type
 of XSS](http://www.webappsec.org/projects/articles/071105.shtml), which
-he coined [DOM Based XSS](DOM_Based_XSS "wikilink"). These 3 types of
+he coined [DOM Based XSS](../attacks/DOM_Based_XSS). These 3 types of
 XSS are defined as follows:
 
-### [Stored XSS](Cross-site_Scripting_\(XSS\)#Stored_XSS_Attacks "wikilink") (AKA Persistent or Type I)
+### [Stored XSS](../attacks/Cross-site-Scripting-(XSS)#stored-xss-attacks) (AKA Persistent or Type I)
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:25]:
```suggestion
### [Stored XSS](attacks/xss/#stored-xss-attacks) (AKA Persistent or Type I)
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 5]
```diff
@@ -155,14 +155,14 @@ writeup), Amit Klein, July 2005
 ### Related OWASP Articles
 
   - [Cross-site Scripting
-    (XSS)](Cross-site_Scripting_\(XSS\) "wikilink")
+    (XSS)](../attacks/Cross-site-Scripting-(XSS))
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:158]:
```suggestion
    (XSS)](attacks/xss/)
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 6]
```diff
@@ -43,7 +43,7 @@ and without permanently storing the user provided data. In some cases,
 the user provided data may never even leave the browser (see DOM Based
 XSS next).
 
-### [DOM Based XSS](DOM_Based_XSS "wikilink") (AKA Type-0)
+### [DOM Based XSS](../attacks/DOM_Based_XSS) (AKA Type-0)
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:46]:
```suggestion
### [DOM Based XSS](attacks/DOM_Based_XSS) (AKA Type-0)
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 7]
```diff
@@ -155,14 +155,14 @@ writeup), Amit Klein, July 2005
 ### Related OWASP Articles
 
   - [Cross-site Scripting
-    (XSS)](Cross-site_Scripting_\(XSS\) "wikilink")
+    (XSS)](../attacks/Cross-site-Scripting-(XSS))
   - [Stored
-    XSS](Cross-site_Scripting_\(XSS\)#Stored_XSS_Attacks "wikilink")
+    XSS](../attacks/Cross-site-Scripting-(XSS)#stored-xss-attacks)
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:160]:
```suggestion
    XSS](attacks/xss/#stored-xss-attacks)
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 8]
```diff
@@ -155,14 +155,14 @@ writeup), Amit Klein, July 2005
...
     (AKA Persistent or Type I XSS)
   - [Reflected
-    XSS](Cross-site_Scripting_\(XSS\)#Reflected_XSS_Attacks "wikilink")
+    XSS](../attacks/Cross-site-Scripting-(XSS)#reflected-xss-attacks)
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:163]:
```suggestion
    XSS](attacks/xss/#reflected-xss-attacks)
```

File: pages/Types_of_Cross-Site_Scripting.md [hunk 9]
```diff
@@ -155,14 +155,14 @@ writeup), Amit Klein, July 2005
...
     (AKA Non-Persistent or Type II XSS)
-  - [DOM Based XSS](DOM_Based_XSS "wikilink")
+  - [DOM Based XSS](../attacks/DOM_Based_XSS)
```
Comment (kingthorin) [inline pages/Types_of_Cross-Site_Scripting.md:165]:
```suggestion
  - [DOM Based XSS](attacks/DOM_Based_XSS)
```

File: pages/Types_of_Cross-Site_Scripting.md (hunk 1 above)
Comment (kingthorin) [reply pages/Types_of_Cross-Site_Scripting.md:19]: It's https://owasp.org/www-community/attacks/xss/ due to the permalink in it's front mater.

**Comment by adamczi** [comment]
Thanks! Sorry for bothering you with those `../`s 
This resolves (I think) #26

**Pull request merged**